### PR TITLE
Cleanup sockets for process that are not running

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
This PR is going to reduce memory usage by  eBPF.plugin, and help us to fix https://github.com/netdata/netdata/issues/12027.

We are also removing a `kprobe` that we won't use.
##### Test Plan

1. Clone this Branch.
4. Get binaries compiled from this [link](https://github.com/netdata/kernel-collector/actions/runs/1996028994)
5. Extract the files in a specific directory:

```bash
# for i in `ls *.zip`; do unzip $i; rm .gitkeep ; rm $i; done
# for i in `ls *.xz`; do tar -xf $i; rm $i* ; done
```
6. Compile tester 
```bash
# make tester
```
7. Run the tester for socket:

```bash
# ./kernel/legacy_test --netdata-path PATH_TO_ARTIFACTS --content --iteration 2 --socket --log-path FILE.txt
````

##### Additional information
This PR was tested on:

| Linux Distribution | kernel version | Socket log |
|--------------------|----------------|---------|
| Slackware Current  |     5.16.12    | [slackware_5_16.txt](https://github.com/netdata/kernel-collector/files/8227898/slackware_5_16.txt) |
| Arch Linux         |  5.16.13-arch1 | [arch_5_16.txt](https://github.com/netdata/kernel-collector/files/8227900/arch_5_16.txt) |
| Ubuntu 21.04       |    5.11.0-49   | [ubuntu_5_11.txt](https://github.com/netdata/kernel-collector/files/8227902/ubuntu_5_11.txt) |
| Manjaro 21.1       |    5.10.102-1  | [manjaro_5_10.txt](https://github.com/netdata/kernel-collector/files/8227903/manjaro_5_10.txt) | 
| Ubuntu 18.04       |    5.4.0-104   | [ubuntu_5_4.txt](https://github.com/netdata/kernel-collector/files/8227906/ubuntu_5_4.txt) |
| Ubuntu 18.04       |   4.18.0-25.26 | [ubuntu_4_18.txt](https://github.com/netdata/kernel-collector/files/8227905/ubuntu_4_18.txt) |
| Ubuntu 18.04       |   4.15.0-171   | [ubuntu_4_15.txt](https://github.com/netdata/kernel-collector/files/8227904/ubuntu_4_15.txt) | 
| Rocky 8.5          | 4.18.0-348.12.2.el8_5 | [rocky_4_18.txt](https://github.com/netdata/kernel-collector/files/8227899/rocky_4_18.txt) |
| CentOS 7.9         | 3.10.0-1160.53.1.el7.x86_64. |  [centos_3_10.txt](https://github.com/netdata/kernel-collector/files/8227901/centos_3_10.txt) |

### CODE REVIEW

File: kernel/socket_kern.c [hunk 1]
```diff
@@ -765,5 +765,43 @@ int trace_udp_sendmsg(struct pt_regs* ctx)
...
+/************************************************************************************
+ *
+ *                           CLEAN UP SECTION
+ *
+ ***********************************************************************************/
+
+/**
+ * Release task socket
+ *
+ * To remove the socket when it is not more necessary help us to reduce the default
```
Comment (vlvkobal) [inline kernel/socket_kern.c:777]:
```suggestion
 * Removing a socket when it's no longer needed helps us reduce the default
```

File: kernel/socket_kern.c [hunk 2]
```diff
@@ -765,5 +765,43 @@ int trace_udp_sendmsg(struct pt_regs* ctx)
...
+ * size used with our tables.
+ *
+ * When a process close to fast that apps.plugin or cgroup.plugin cannot detect,
+ * we will not show information for them, but we will have global information. So to remove
+ * socket information it is not so hard.
```
Comment (vlvkobal) [inline kernel/socket_kern.c:782]:
```suggestion
 * When a process stops so fast that apps.plugin or cgroup.plugin cannot detect it, we don't show
 * the information about the process, so it is safe to remove the information about the socket.
```

**Review by vlvkobal — APPROVED** [review]

**Review by vlvkobal — APPROVED** [review]

**Comment by vlvkobal** [comment]
Why did you revert the changes?

**Comment by thiagoftsm** [comment]
> Why did you revert the changes?

CI was not running, because PR https://github.com/netdata/kernel-collector/pull/296 was merged, so I had to push another commit. I did not change any code.

**Comment by vlvkobal** [comment]
You removed the latest two commits with fixes to the comments.

**Comment by thiagoftsm** [comment]
> You removed the latest two commits with fixes to the comments.

Fixed, I did not update the local before to rebase.

**Review by vlvkobal — APPROVED** [review]

**Review by vlvkobal — APPROVED** [review]

**Review by erdem2000 — APPROVED** [review]

**Pull request merged**